### PR TITLE
feat(replay): details issue tab support backend errors

### LINKED ISSUE #48478: Replay Details > Issues tab should reflect backend errors

### PR DESCRIPTION
**Pull request by eliashussary**
## Summary 
Adds backend error support in replay details issue tab.

Requires: https://github.com/getsentry/sentry/pull/48493
Relates to: https://github.com/getsentry/sentry/issues/48249
Closes: https://github.com/getsentry/sentry/issues/48478

image

### CODE REVIEW

File: static/app/views/replays/detail/issueList.tsx [hunk 1]
```diff
@@ -54,8 +51,6 @@ function IssueList({projectId, replayId}: Props) {
         `/organizations/${organization.slug}/issues/`,
         {
           query: {
-            // TODO(replays): What about backend issues?
-            project: projectId,
```
Comment (eliashussary) [inline static/app/views/replays/detail/issueList.tsx:58]: @JoshFerge pretty sure we're gonna run into the `global-views` gating here. 

I don't think i can ship this unless there's a way around this.

File: static/app/api.tsx [hunk 2]
```diff
@@ -217,6 +217,10 @@ export type RequestOptions = RequestCallbacks & {
    * Values to attach to the body of the request.
    */
   data?: any;
+  /**
+   * Headers add to the request.
+   */
+  headers?: Record<string, any>;
```
Comment (ryan953) [inline static/app/api.tsx:223]:
```suggestion
  headers?: Record<string, string>;
```

File: static/app/views/replays/detail/issueList.tsx [hunk 3]
```diff
@@ -54,10 +51,11 @@ function IssueList({projectId, replayId}: Props) {
...
             query: `replayId:${replayId}`,
           },
+          headers: {
+            'x-sentry-replay-request': 1,
```
Comment (ryan953) [inline static/app/views/replays/detail/issueList.tsx:57]:
```suggestion
            'x-sentry-replay-request': '1',
```

**Review by ryan953 — APPROVED** [review]

File: static/app/api.tsx (hunk 2 above)
Comment (eliashussary) [reply static/app/api.tsx]: caught a copypasta

File: static/app/views/replays/detail/issueList.tsx (hunk 1 above)
Comment (eliashussary) [reply static/app/views/replays/detail/issueList.tsx:58]: fixed by: https://github.com/getsentry/sentry/pull/48493

File: static/app/api.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/api.tsx]: haha, as i was writing about it.... was able to draw from my own very recent examples

**Pull request merged**